include/eve/module/core/regular/gather.hpp

```diff
@@ -12,19 +12,23 @@
 namespace eve
 {
   template<typename Options>
-  struct gather_t : callable<gather_t, Options, conditional_option>
+  struct gather_t : conditional_callable<gather_t, Options>
   {
     template<arithmetic_value T, integral_value U>
     constexpr EVE_FORCEINLINE as_wide_as_t<T, U> operator()(T const* ptr, U idx) const noexcept
     {
-      static_assert(simd_value<U> || !Options::contains(condition_key), "[eve::gather] Scalar values can't be masked by SIMD logicals");
+      static_assert(detail::validate_mask_for<decltype(this->options()), as_wide_as_t<T, U>>(),
```

Comment:
We don't do the if constexpr here?
Like we had a discussion that it's better for error messages?

Also - won't you get this validation from conditional?